### DISCUSSION TITLE
Range Component: Centered labels on the slider

### DISCUSSION THREAD

**Discussion by fredbradley** [saadeghi/daisyui#3234] [Q&A] [answered]
Hello, 

I love the range component - but is anyone able to help me style the labels so that they are central to where the component block is on each step? 

Image for example: 
![image](https://github.com/user-attachments/assets/b8ba7340-896c-499c-9b10-74823a3c0056)

Even when the slider is at 'Possibly Leaving' the circle isn't actually central on the line, and as you can see - the ends aren't the actual ends of the slider. 
![image](https://github.com/user-attachments/assets/42e0a2ba-e9c1-4a81-8c7f-45b5e1eb0f81)
![image](https://github.com/user-attachments/assets/2d5c7d97-85b5-407f-8f1f-6acc244d64c8)

Any help appreciated,

**Comment by saadeghi** [comment]
Can you share the code on Tailwind Play?

**Reply by fredbradley** [reply]
https://play.tailwindcss.com/fumaRlfQRa

I think it's to do with the length of text being used as the descriptors below. 

Do you think there's any solve? Would there be a way to start from the middle, and resize the range so that it starts and finishes only at the centre of the text in the first and last `span`?

**Reply by saadeghi** [reply, marked as answer]
Here's an example: https://play.tailwindcss.com/GMTyPd5pql

I used grid instead of flex so all 5 columns get equal width. We want the range thumb to be aligned with the line from first item and last item, and there are 5 columns which means we need 10% horizontal margin for the range + the size of thumb (which is 32px). So `w-[calc(80%+32px)]` would make range fit exactly to the scale.

Let me know if you have a question.

**Reply by fredbradley** [reply]
Genius. Thank you!
